modules/local/isolate_ids_from_kraken2_to_blastn.nf

```diff
@@ -60,14 +60,10 @@ process ISOLATE_IDS_FROM_KRAKEN2_TO_BLASTN {
                 sum_to_filter = sum([lca_mapping[id_] for id_ in tax2filter])
                 unclassified = lca_mapping[0]
 
-                # Add pseudo count
-                if unclassified == 0:
-                    unclassified = 1
-
                 if (
                     sum_to_filter > sum_to_keep
                     and sum_to_filter > $params.cutoff_tax2filter
-                    and sum_to_filter/unclassified > $params.cutoff_unclassified
+                    and sum_to_filter/(unclassified + sum_to_filter) > $params.cutoff_unclassified
```

Comment:
All fine and well, and maybe I misunderstand, but what about kmers that are neither `unclassified` nor to `sum_to_filter`, i.e. `sum_to_keep`? Isnt the total number of kmers `unclassified + sum_to_filter + sum_to_keep`?
In order to achieve a 0 to 1 ratio in the sense of `minimum fraction of kmers to filter of total kmers`, it should be `sum_to_filter/(unclassified + sum_to_filter + sum_to_keep)` ?
Therefore
```suggestion
                    and sum_to_filter/(unclassified + sum_to_filter + sum_to_keep) > $params.cutoff_unclassified
```

Reply:
The model was rewritten and is now:

$\sum(toFilter)> cutoffTax2filter$  
$\wedge$ 
$\frac{\sum(toFilter)}{\sum(toKeep)+\sum(toFilter)}> cutoffTax2keep$
$\wedge$ 
$\frac{\sum(toFilter)}{\sum(toFilter)+unclassified} > cutoffUnclassified$ 

If this evaluates to `True` the read is classified as to filter/to assess for.